src/common/com/intellij/plugins/haxe/lang/psi/HaxeClassResolveCache.java

```diff
@@ -22,17 +23,18 @@
 import com.intellij.openapi.project.Project;
 import com.intellij.psi.impl.AnyPsiChangeListener;
 import com.intellij.psi.impl.PsiManagerImpl;
-import com.intellij.util.containers.ConcurrentWeakHashMap;
 import com.intellij.util.containers.ContainerUtil;
 import com.intellij.util.messages.MessageBus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.concurrent.ConcurrentMap;
+
 /**
  * @author: Fedor.Korotkov
  */
 public class HaxeClassResolveCache {
-  private final ConcurrentWeakHashMap<HaxeClass, HaxeClassResolveResult> myMap = createWeakMap();
+  private final ConcurrentMap<HaxeClass, HaxeClassResolveResult> myMap = ContainerUtil.createConcurrentWeakMap();
```

Comment:
Why this change?

Reply:
ConcurrentWeakHashMap is deprecated and that what you see is recommended for replacement.